com.unity.render-pipelines.universal/Editor/ShaderPreprocessor.cs

```diff
@@ -485,20 +761,67 @@ public void OnPostprocessBuild(BuildReport report)
         public void OnPreprocessBuild(BuildReport report)
         {
             FetchAllSupportedFeatures();
+            FetchAllSupportedFeaturesFromVolumes();
 #if PROFILE_BUILD
             Profiler.enableBinaryLog = true;
             Profiler.logFile = "profilerlog.raw";
             Profiler.enabled = true;
 #endif
         }
 
+        static bool TryGetRenderPipelineAssetsForBuildTarget(BuildTarget buildTarget, List<UniversalRenderPipelineAsset> urps)
```

Comment:
Are we okay with this hack until we will get proper API? As you see there is lots of protections and code will simply return false if anything changed in API fallbacking to old functionality.

Reply:
Have we estimated the work for the proper API?

For 22.1, I think it's fine but the then we should start on the Proper API immediately and update this as soon as that lands so this will not be in the final release. 

For 21.2, I would really want to get the new API backported and then this PR if possible. If not, I can live with it since like you say there are a lot of checks.